ui/pages/confirmations/components/confirm/info/smart-account-switch-info/smart-account-switch-info.tsx

```diff
@@ -0,0 +1,26 @@
+import { TransactionMeta } from '@metamask/transaction-controller';
+import React from 'react';
+
+import { useConfirmContext } from '../../../../context/confirm';
+import { AdvancedDetails } from '../shared/advanced-details/advanced-details';
+import { GasFeesSection } from '../shared/gas-fees-section/gas-fees-section';
+import { TransactionAccountDetails } from '../batch/transaction-account-details';
+
+const SmartAccountSwitchInfo = () => {
+  const { currentConfirmation: transactionMeta } =
+    useConfirmContext<TransactionMeta>();
+
+  if (!transactionMeta?.txParams) {
+    return null;
+  }
+
+  return (
+    <>
+      <TransactionAccountDetails />
```

Comment:
The batch support was intentionally not in it's own `Info` component as it's not a specific transaction type, but rather an extension to any other transaction type.

You could have an approval or transfer transaction, that needs it's specific info, plus info about the upgrade.

Here for example, we're missing the simulation details if you simultaneously upgrade and do something, from a `wallet_sendCalls`.

Reply:
Ah you are right, I made this component for only upgrade / downgrade confirmation and I did not thought of scenario where upgrade will have other nested transaction also. Let me check this more.

Reply:
PR is updated